pkg/controllers/resources/persistentvolumes/syncer.go

```diff
@@ -240,33 +252,46 @@ func (s *persistentVolumeSyncer) SyncToVirtual(ctx *synccontext.SyncContext, pOb
 	return ctrl.Result{}, nil
 }
 
-func (s *persistentVolumeSyncer) shouldSync(ctx context.Context, pObj *corev1.PersistentVolume) (bool, *corev1.PersistentVolumeClaim, error) {
+func (s *persistentVolumeSyncer) shouldSync(ctx context.Context, pObj *corev1.PersistentVolume) (shouldSync bool, vpvc *corev1.PersistentVolumeClaim, ppvc *corev1.PersistentVolumeClaim, err error) {
 	// is there an assigned PVC?
 	if pObj.Spec.ClaimRef == nil {
 		if translate.Default.IsManagedCluster(pObj) {
-			return true, nil, nil
+			return true, nil, nil, nil
 		}
 
-		return false, nil, nil
+		return false, nil, nil, nil
 	}
 
 	vPvc := &corev1.PersistentVolumeClaim{}
-	err := clienthelper.GetByIndex(ctx, s.virtualClient, vPvc, constants.IndexByPhysicalName, pObj.Spec.ClaimRef.Namespace+"/"+pObj.Spec.ClaimRef.Name)
+	err = clienthelper.GetByIndex(ctx, s.virtualClient, vPvc, constants.IndexByPhysicalName, pObj.Spec.ClaimRef.Namespace+"/"+pObj.Spec.ClaimRef.Name)
 	if err != nil {
 		if !kerrors.IsNotFound(err) {
-			return false, nil, err
+			return false, nil, nil, err
 		} else if translate.Default.IsManagedCluster(pObj) {
-			return true, nil, nil
+			return true, nil, nil, nil
 		}
 
+		// this will only work if multi ns isn't enabled, because translator returns an error
+		// as it is not a valid call
+		if s.isMultiNS {
+			return true, nil, nil, nil
+		}
 		namespace, err := translate.Default.LegacyGetTargetNamespace()
 		if err != nil {
-			return false, nil, nil
+			return false, nil, nil, nil
 		}
-		return pObj.Spec.ClaimRef.Namespace == namespace && pObj.Spec.PersistentVolumeReclaimPolicy == corev1.PersistentVolumeReclaimRetain, nil, nil
+		return pObj.Spec.ClaimRef.Namespace == namespace && pObj.Spec.PersistentVolumeReclaimPolicy == corev1.PersistentVolumeReclaimRetain, nil, nil, nil
+	}
+
+	pPvc := &corev1.PersistentVolumeClaim{}
+	err = s.physicalClient.Get(ctx, types.NamespacedName{Namespace: pObj.Spec.ClaimRef.Namespace, Name: pObj.Spec.ClaimRef.Name}, pPvc)
```

Comment:
Shouldn't we validate here that the ClaimRef.Namespace actually matches the namespace we are working in as vCluster?

Reply:
e.g. this could cause problematic errors if we cannot list in that namespace, which would make everything fail as a non not found error is thrown

Reply:
but isn't that what we want though? if the pv has a claimref with a pvc that we don't have access to, we don't own the pv